### PR TITLE
Add references to Cognito UserPool

### PR DESCRIPTION
**Pull request by mbbush**
### Description of your changes

Adds references to cognito user pool, and makes the example uptestable. I couldn't use the Object.s3 resource because of https://github.com/upbound/provider-aws/issues/978, but the (deprecated in the terraform provider) BucketObject.s3 resource works fine. 

I have:

- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

`make e2e UPTEST_EXAMPLE_LIST=examples/cognitoidp/userpool.yaml`

### CODE REVIEW

**Comment by mbbush** [comment]
/test-examples="examples/s3/object.yaml"

**Comment by mbbush** [comment]
/test-examples="examples/cognito/userpool.yaml"

**Comment by mbbush** [comment]
/test-examples="examples/cognitoidp/userpool.yaml"

File: config/cognitoidp/config.go [hunk 1]
```diff
@@ -52,5 +54,68 @@ func Configure(p *config.Provider) {
 				"email_verification_subject",
 			},
 		}
+		r.References["email_configuration.configuration_set"] = config.Reference{
+			TerraformName: "aws_ses_configuration_set",
+		}
+		r.References["email_configuration.source_arn"] = config.Reference{
+			TerraformName: "aws_ses_email_identity",
+			Extractor:     common.PathARNExtractor,
+		}
```
Comment (mbbush) [inline config/cognitoidp/config.go:63]: So it turns out there are actually two terraform resources for these objects, these and the corresponding `aws_sesv2_` ones. As far as I can tell, they are simply two different interfaces for the same objects, with the same ARNs. I'm conflicted about what to do here, but perhaps these would be better left off?

**Comment by mbbush** [comment]
/test-examples="examples/cognitoidp/userpool.yaml"

**Comment by turkenf** [comment]
/test-examples="examples/cognitoidp/userpool.yaml"

File: config/cognitoidp/config.go (hunk 1 above)
Comment (turkenf) [reply config/cognitoidp/config.go]: It seems more logical not to define a reference for these for now.

**Review by turkenf — APPROVED** [review]
Thank you @mbbush, LGTM.

**Pull request merged**